### PR TITLE
Fix bogus message "outdated scancode-toolkit version"

### LINKED ISSUE #2796: Bogus "Outdated ScanCode Toolkit version" message when running ScanCode 30.1.0
### How To Reproduce

```
ort (master) $ scancode --license LICENSE --json-pp a.json
Setup plugins...
Collect file inventory...
Scan files for: licenses with 1 process(es)...
[####################] 2                 
Scanning done.
Summary:        licenses with 1 process(es)
Errors count:   0
Scan Speed:     1.50 files/sec. 
Initial counts: 1 resource(s): 1 file(s) and 0 directorie(s) 
Final counts:   1 resource(s): 1 file(s) and 0 directorie(s) 
Timings:
  scan_start: 2022-01-14T092000.290009
  scan_end:   2022-01-14T092007.659667
  setup_scan:licenses: 6.70s
  setup: 6.70s
  scan: 0.67s
  total: 7.37s
Removing temporary files...done.
WARNING: Outdated ScanCode Toolkit version! You are using an outdated version of ScanCode Toolkit: 30.1.0 released on: 2021-09-24. A new version is available with important improvements including bug and security fixes, updated license, copyright and package detection, and improved scanning accuracy. Please download and install the latest version of ScanCode. Visit https://github.com/nexB/scancode-toolkit/releases for details.
```

So ScanCode version 30.1.0 complains about being outdated although there currently is no newer release at https://github.com/nexB/scancode-toolkit/releases or https://pypi.org/project/scancode-toolkit/#history.

### PR DESCRIPTION
**Pull request by adityasangave**
Fixes #2796 

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁

Signed-off-by: Author Name <adityasangave2112@gmail.com>

### CODE REVIEW

File: src/scancode/outdated.py [hunk 1]
```diff
@@ -143,7 +143,7 @@ def check_scancode_version_locally(
     Return a message to display if outdated or None. Work offline, without a
     PyPI remote check.
     """
-    if is_outdated(release_date):
+    if is_outdated(release_date) and installed_version<fetch_latest_version():
```
Comment (pombredanne) [inline src/scancode/outdated.py:146]: That's smart! We need still to make sure that this is NOT called when offline. 
This is breaking the expectation laid above https://github.com/nexB/scancode-toolkit/pull/2800/files#diff-ce07c0640f6e9f4595b760563228f57f721c3c142fecbdd5a33b6efcbb2b9e3cR143

Comment (sschuberth) [reply src/scancode/outdated.py:146]: I also believe the check is wrong at this place. IMO we should have something like a wrapping `check_scancode_version` function that calls `check_scancode_version_locally` when offline and `check_scancode_version_remotely` when online.

Comment (adityasangave) [reply src/scancode/outdated.py:146]: Ok I'll make the changes

Comment (pombredanne) [reply src/scancode/outdated.py:146]: Thanks!
I suggest we discuss a little in https://github.com/nexB/scancode-toolkit/issues/2796 to find the correct design.

Comment (adityasangave) [reply src/scancode/outdated.py:146]: Yess i was thinking the same becasue i was unable to think any solution for fixing the problem locally

**Pull request closed without merging**